core/src/main/java/org/mule/registry/TransientRegistry.java

```diff
@@ -283,7 +289,17 @@ public void registerObject(String key, Object object, Object metadata) throws Re
             return;
         }
 
-        registryMap.putAndLogWarningIfDuplicate(key, object);
+        doRegisterObject(key, object, metadata);
+    }
+
+    protected <T> T doGet(String key)
```

Comment:
i will change the registryMap to no longer be accessible so this comment won't apply anymore